agave_app/AppearanceSettingsWidget.cpp

```diff
@@ -351,26 +316,134 @@ QAppearanceSettingsWidget::QAppearanceSettingsWidget(QWidget* pParent,
+                                         "Enable/disable clip plane" };
+  m_clipPlaneSection = new Section("Clip Plane", 0, &checkBoxInfo);
+  // section checkbox turns clip plane on or off
+  QObject::connect(m_clipPlaneSection, &Section::checked, [this](bool is_checked) {
+    if (this->m_scene && this->m_scene->m_clipPlane) {
+      this->m_scene->m_clipPlane->m_enabled = is_checked;
+      m_qrendersettings->renderSettings()->m_DirtyFlags.SetFlag(RoiDirty);
+      emit this->m_qrendersettings->Selected(
+        is_checked && !m_hideUserClipPlane->isChecked() ? this->m_scene->m_clipPlane.get() : nullptr);
+    }
+  });
+
+  auto* sectionLayout = Controls::createAgaveFormLayout();
+
+  auto btnLayout = new QHBoxLayout();
+
+  m_clipPlaneRotateButton = new QPushButton("Rotate");
+  m_clipPlaneRotateButton->setStatusTip(tr("Show interactive controls in viewport for clip plane rotation angle"));
+  m_clipPlaneRotateButton->setToolTip(tr("Show interactive controls in viewport for clip plane rotation angle"));
+  btnLayout->addWidget(m_clipPlaneRotateButton);
+  QObject::connect(m_clipPlaneRotateButton, &QPushButton::clicked, [this, pToggleRotateAction]() {
+    if (!this->m_scene) {
+      return;
+    }
+    // if we were already selected AND already in rotate mode, then this should switch off rotate mode.
+    if (this->m_scene->m_selection == this->m_scene->m_clipPlane.get() && pToggleRotateAction->isChecked()) {
+      emit this->m_qrendersettings->Selected(nullptr);
+      pToggleRotateAction->trigger();
+      if (!m_hideUserClipPlane->isChecked()) {
+        emit this->m_qrendersettings->Selected(this->m_scene->m_clipPlane.get());
+      }
+    } else {
+      emit this->m_qrendersettings->Selected(this->m_scene->m_clipPlane.get());
+      pToggleRotateAction->trigger();
+    }
+  });
+
+  m_clipPlaneTranslateButton = new QPushButton("Translate");
+  m_clipPlaneTranslateButton->setStatusTip(tr("Show interactive controls in viewport for clip plane translation"));
+  m_clipPlaneTranslateButton->setToolTip(tr("Show interactive controls in viewport for clip plane translation"));
+  btnLayout->addWidget(m_clipPlaneTranslateButton);
+  QObject::connect(m_clipPlaneTranslateButton, &QPushButton::clicked, [this, pToggleTranslateAction]() {
+    if (!this->m_scene) {
+      return;
+    }
+    // if we were already selected AND already in translate mode, then this should switch off translate mode.
+    if (this->m_scene->m_selection == this->m_scene->m_clipPlane.get() && pToggleTranslateAction->isChecked()) {
+      emit this->m_qrendersettings->Selected(nullptr);
+      pToggleTranslateAction->trigger();
+      if (!m_hideUserClipPlane->isChecked()) {
+        emit this->m_qrendersettings->Selected(this->m_scene->m_clipPlane.get());
+      }
+    } else {
+      emit this->m_qrendersettings->Selected(this->m_scene->m_clipPlane.get());
+      pToggleTranslateAction->trigger();
+    }
+  });

```

Comment:
Looks like this Translate button click handler has similar logic to the Rotate button(L344-359), I'm wondering if we could make them DRYer?